index.bs

```diff
@@ -0,0 +1,603 @@
+```
+</div>
+
+See [[PERMISSIONS-POLICY]] for more details.
+</div>
+
+## The {{IdleDetector}} interface ## {#api-idledetector}
+
+<xmp class="idl">
+dictionary IdleOptions {
+  [EnforceRange] unsigned long long threshold;
+  AbortSignal signal;
+};
+
+[
+  SecureContext,
+  Exposed=(Window,DedicatedWorker)
+] interface IdleDetector : EventTarget {
+  constructor();
+  readonly attribute UserIdleState? userState;
+  readonly attribute ScreenIdleState? screenState;
+  attribute EventHandler onchange;
+  [Exposed=Window] static Promise<PermissionState> requestPermission();
+  Promise<void> start(optional IdleOptions options = {});
+};
+</xmp>
+
+Instances of {{IdleDetector}} are created with the [=internal slots=] described
+in the following table:
+
+<table dfn-for=IdleDetector dfn-type=attribute>
+  <tr>
+    <th>[=Internal slot=]</th>
+    <th>Initial value</th>
+    <th>Description (non-normative)</th>
+  </tr>
+  <tr>
+    <td><dfn>\[[state]]</dfn></th>
+    <td>`"stopped"`</td>
+    <td>Tracks the active state of the {{IdleDetector}}</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[threshold]]</dfn></th>
+    <td>`undefined`</td>
+    <td>The configured idle detection threshold</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[userState]]</dfn></td>
+    <td>`null`</td>
+    <td>The last known user idle state</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[screenState]]</dfn></td>
+    <td>`null`</td>
+    <td>The last known screen idle state</td>
+  </tr>
+</table>
+
+<wpt>
+  idlharness-worker.https.window.js
+  idlharness.https.window.js
+</wpt>
+
+Methods on this interface typically complete asynchronously, queuing work on the
+<dfn>idle detection task source</dfn>.
+
+### {{IdleDetector/userState}} attribute ### {#api-idledetector-userstate}
+
+<div algorithm>
+The <dfn attribute for=IdleDetector>userState</dfn> getter steps are:
+
+1.  Return [=this=].{{[[userState]]}}.
+
+</div>
+
+<div algorithm="user becomes idle">
+
+For each {{IdleDetector}} instance |detector| the [=user agent=] MUST
+continuously monitor the following conditions:
+
+*   |detector|.{{[[state]]}} is `"started"`.
+*   |detector|.{{[[userState]]}} is {{"active"}}.
+*   The user has not interacted with the device is at least
+    |detector|.{{[[threshold]]}} milliseconds.
+
+If all of these conditions are true the [=user agent=] MUST [=queue a global
+task=] on the [=relevant global object=] of |detector| using the [=idle
+detection task source=] to run the following steps:
+
+1.  Set |detector|.{{[[userState]]}} to {{"idle"}}.
+1.  [=Fire an event=] named {{"change"}} at |detector|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+<div algorithm="user becomes active">
+
+For each {{IdleDetector}} instance |detector| the [=user agent=] MUST
+continuously monitor the following conditions:
+
+*   |detector|.{{[[state]]}} is `"started"`.
+*   |detector|.{{[[userState]]}} is {{"idle"}}.
+*   The user has interacted with the device is at least
+    |detector|.{{[[threshold]]}} milliseconds.
```

Comment:
Sentence seems ungrammatical... I'm not sure when this is supposed to fire, actually. Maybe it's "The user has interacted with the device within the last detector.[[threshold]] milliseconds?" But I'm not sure why threshold is involved here.

If the monitoring is continuous, then maybe instead it should be something like "When the user interacts with the device, if state is started and userState is active, then..."

Reply:
Your assumption is correct, it should be "within the last". The threshold allows the site to specify how aggressively they want to monitor the user's activity state, and must be at least a minute to prevent the site from observing too much information about input events in other contexts. You're right that for the idle to active transition I could also write this as simply "when the user interacts with the device" and get the same effect. I've reduced some of the repetition by combining all these algorithms into a single one which I think expresses the transitions more clearly.